### PR TITLE
Use cubical path in library

### PR DESCRIPTION
**Pull request by ice1000**
@imkiva help me with path eta

### CODE REVIEW

**Comment by ice1000** [comment]
```
����: Internal error: java.lang.UnsupportedOperationException: TODO
java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: TODO
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1770)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(CompletableFuture.java:1760)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.UnsupportedOperationException: TODO
	at org.aya.tyck.unify.DefEq.doCompareTyped(DefEq.java:305)
	at org.aya.tyck.unify.DefEq.compare(DefEq.java:122)
	at org.aya.tyck.unify.DefEq.doCompareUntyped(DefEq.java:370)
	at org.aya.tyck.unify.DefEq.compareUntyped(DefEq.java:136)
	at org.aya.tyck.unify.DefEq.compare(DefEq.java:120)
	at org.aya.tyck.unify.DefEq.compare(DefEq.java:109)
	at org.aya.tyck.ExprTycker.unifyTy(ExprTycker.java:707)
	at org.aya.tyck.ExprTycker.unifyTyMaybeInsert(ExprTycker.java:741)
	at org.aya.tyck.ExprTycker.doInherit(ExprTycker.java:462)
	at org.aya.tyck.ExprTycker.inherit(ExprTycker.java:559)
	at org.aya.tyck.ExprTycker.doSynthesize(ExprTycker.java:190)
	at org.aya.tyck.ExprTycker.synthesize(ExprTycker.java:566)
	at org.aya.tyck.ExprTycker.doInherit(ExprTycker.java:462)
	at org.aya.tyck.ExprTycker.inherit(ExprTycker.java:559)
	at org.aya.tyck.StmtTycker.doSimpleFn(StmtTycker.java:169)
	at org.aya.tyck.StmtTycker.lambda$simpleFn$7(StmtTycker.java:162)
	at org.aya.tyck.StmtTycker.traced(StmtTycker.java:57)
	at org.aya.tyck.StmtTycker.simpleFn(StmtTycker.java:162)
	at org.aya.tyck.order.AyaSccTycker.checkSimpleFn(AyaSccTycker.java:154)
	at org.aya.tyck.order.AyaSccTycker.checkUnit(AyaSccTycker.java:122)
	at org.aya.tyck.order.AyaSccTycker.tyckSCC(AyaSccTycker.java:64)
	at org.aya.util.tyck.OrgaTycker.tyckSCC(OrgaTycker.java:25)
	at kala.collection.immutable.ImmutableVectors.forEachRec(ImmutableVectors.java:745)
	at kala.collection.immutable.ImmutableVectors$BigVector.forEach(ImmutableVectors.java:567)
	at org.aya.resolve.module.ModuleLoader.tyckModule(ModuleLoader.java:41)
	at org.aya.cli.library.LibraryModuleLoader.load(LibraryModuleLoader.java:79)
	at org.aya.resolve.module.CachedModuleLoader.lambda$load$0(CachedModuleLoader.java:32)
	at kala.collection.mutable.MutableTreeMap.getOrPut(MutableTreeMap.java:483)
	at org.aya.resolve.module.CachedModuleLoader.load(CachedModuleLoader.java:32)
	at org.aya.resolve.module.ModuleLoader.load(ModuleLoader.java:63)
	at org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckOne(LibraryCompiler.java:316)
	at org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckSCC(LibraryCompiler.java:305)
	at org.aya.util.tyck.OrgaTycker.tyckSCC(OrgaTycker.java:25)
	at kala.function.CheckedConsumer.accept(CheckedConsumer.java:46)
	at kala.collection.immutable.ImmutableVectors$Vector1.forEach(ImmutableVectors.java:440)
	at kala.collection.base.Traversable.forEachChecked(Traversable.java:684)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:209)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:179)
	at org.aya.cli.library.LibraryCompiler.make(LibraryCompiler.java:158)
	at org.aya.cli.utils.AyaCompiler.catching(AyaCompiler.java:30)
	at org.aya.cli.library.LibraryCompiler.start(LibraryCompiler.java:144)
	at org.aya.lsp.server.AyaService.loadLibrary(AyaService.java:155)
	at kala.collection.internal.view.SeqViews$FlatMapped.lambda$iterator$0(SeqViews.java:1142)
	at kala.collection.base.Iterators$3.next(Iterators.java:748)
	at kala.collection.base.Iterators$ConcatAll.hasNext(Iterators.java:2056)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:154)
	at kala.collection.immutable.ImmutableVector.from(ImmutableVector.java:139)
	at kala.collection.CollectionLike.toImmutableVector(CollectionLike.java:104)
	at kala.collection.CollectionLike.toImmutableSeq(CollectionLike.java:92)
	at org.aya.lsp.server.AyaService.reload(AyaService.java:146)
	at org.aya.lsp.server.AyaServer.lambda$load$0(AyaServer.java:37)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
```

**Comment by ice1000** [comment]
```
D:\git-repos\aya-prover [coe-subst ≡]> aya --remake $PWD\base\src\test\resources\success\
Compiling common
  [Info] Resolving source file dependency
    Done in 477ms
  [Tyck] Primitives (Primitives.aya)
  [Tyck] Paths (Paths.aya)
  [Tyck] Homotopy (Homotopy.aya)
  [Tyck] Data::Unit (Data\Unit.aya)
  [Tyck] Data::Empty (Data\Empty.aya)
  [Tyck] Data::Bool (Data\Bool.aya)
org.aya.generic.util.InternalException: j
        at aya.base/org.aya.tyck.env.LocalCtx.get(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.doCompareUntyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.compareUntyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.doCompareTyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.compare(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.doCompareUntyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.compareUntyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.doCompareTyped(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.compare(Unknown Source)
        at aya.base/org.aya.tyck.unify.DefEq.compare(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.boundary(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.lambda$checkBoundaries$12(Unknown Source)
        at aya.guest.cubical/org.aya.guest0x0.cubical.CofThy.vdash(Unknown Source)
        at aya.guest.cubical/org.aya.guest0x0.cubical.CofThy.conv(Unknown Source)
        at aya.guest.cubical/org.aya.guest0x0.cubical.CofThy.conv(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.lambda$checkBoundaries$13(Unknown Source)
        at kala.collection/kala.collection.IndexedSeqLike.allMatch(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.checkBoundaries(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.doInherit(Unknown Source)
        at aya.base/org.aya.tyck.ExprTycker.inherit(Unknown Source)
        at aya.base/org.aya.tyck.StmtTycker.doSimpleFn(Unknown Source)
        at aya.base/org.aya.tyck.StmtTycker.lambda$simpleFn$7(Unknown Source)
        at aya.base/org.aya.tyck.StmtTycker.traced(Unknown Source)
        at aya.base/org.aya.tyck.StmtTycker.simpleFn(Unknown Source)
        at aya.base/org.aya.tyck.order.AyaSccTycker.checkSimpleFn(Unknown Source)
        at aya.base/org.aya.tyck.order.AyaSccTycker.checkUnit(Unknown Source)
        at aya.base/org.aya.tyck.order.AyaSccTycker.tyckSCC(Unknown Source)
        at aya.util/org.aya.util.tyck.OrgaTycker.tyckSCC(Unknown Source)
        at kala.collection/kala.collection.immutable.ImmutableVectors$Vector1.forEach(Unknown Source)
        at aya.base/org.aya.resolve.module.ModuleLoader.tyckModule(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryModuleLoader.load(Unknown Source)
        at aya.base/org.aya.resolve.module.CachedModuleLoader.lambda$load$0(Unknown Source)
        at kala.collection/kala.collection.mutable.MutableTreeMap.getOrPut(Unknown Source)
        at aya.base/org.aya.resolve.module.CachedModuleLoader.load(Unknown Source)
        at aya.base/org.aya.resolve.module.ModuleLoader.load(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckOne(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler$LibrarySccTycker.tyckSCC(Unknown Source)
        at aya.util/org.aya.util.tyck.OrgaTycker.tyckSCC(Unknown Source)
        at kala.base/kala.function.CheckedConsumer.accept(Unknown Source)
        at kala.collection/kala.collection.immutable.ImmutableVectors$Vector1.forEach(Unknown Source)
        at kala.base/kala.collection.base.Traversable.forEachChecked(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler.make(Unknown Source)
        at aya.cli/org.aya.cli.utils.AyaCompiler.catching(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler.start(Unknown Source)
        at aya.cli/org.aya.cli.library.LibraryCompiler.compile(Unknown Source)
        at aya.cli/org.aya.cli.Main.call(Unknown Source)
        at aya.cli/org.aya.cli.Main.call(Unknown Source)
        at info.picocli/picocli.CommandLine.executeUserObject(Unknown Source)
        at info.picocli/picocli.CommandLine.access$1300(Unknown Source)
        at info.picocli/picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(Unknown Source)
        at info.picocli/picocli.CommandLine$RunLast.handle(Unknown Source)
        at info.picocli/picocli.CommandLine$RunLast.handle(Unknown Source)
        at info.picocli/picocli.CommandLine$AbstractParseResultHandler.execute(Unknown Source)
        at info.picocli/picocli.CommandLine$RunLast.execute(Unknown Source)
        at info.picocli/picocli.CommandLine.execute(Unknown Source)
        at aya.cli/org.aya.cli.Main.main(Unknown Source)
j
Please report the stacktrace to the developers so a better error handling could be made.
Don't forget to inform the version of Aya you're using and attach your code for reproduction.
```


@imkiva 99 me

**Comment by ice1000** [comment]
image

The metas are inserted wrongly

**Comment by imkiva** [comment]
> image
> 
> The metas are inserted wrongly

what is this UI?

**Comment by ice1000** [comment]
@imkiva This is the output of git diff

**Comment by ice1000** [comment]
@imkiva I used cargo install git-delta

File: base/src/main/java/org/aya/tyck/ExprTycker.java [hunk 1]
```diff
@@ -180,7 +180,7 @@ public final class ExprTycker extends Tycker {
             if (argLicit || argument.name() != null) {
               // that implies paramLicit == false
               var holeApp = mockArg(pi.param().subst(subst), argument.expr().sourcePos());
-              app = CallTerm.make(app, holeApp);
+              app = CallTerm.make(app, holeApp); // TODO: cube.makeApp? possible?
```
Comment (ice1000) [inline base/src/main/java/org/aya/tyck/ExprTycker.java:183]: Impossible

**Review by imkiva — DISMISSED** [review]
WE MADE IT, FINALLY!!!

image

**Comment by imkiva** [comment]
The only thing left is merging https://github.com/ice1000/guest0x0/pull/5 and publishing!

**Comment by ice1000** [comment]
> Merging [#505](https://codecov.io/gh/aya-prover/aya-dev/pull/505?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aya-prover) ([a485b16](https://github.com/aya-prover/aya-dev/commit/a485b16da4255a191dc6f6583094792ca2c3cabd)) into [main](https://codecov.io/gh/aya-prover/aya-dev/commit/ada52e07632ed29b22b77a1cb37d58d4a5d765c1?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=aya-prover) ([ada52e0](https://github.com/aya-prover/aya-dev/commit/ada52e07632ed29b22b77a1cb37d58d4a5d765c1)) will **increase** coverage by `0.14%`.

Impressive!!

**Review by ice1000 — COMMENTED** [review]
bors r+

**Comment by ice1000** [comment]
image

**Pull request merged**